jet/job-management/src/main/java/com/hazelcast/samples/jet/jobmanagement/JobSuspendResumeListener.java

```diff
@@ -0,0 +1,72 @@
+package com.hazelcast.samples.jet.jobmanagement;
+
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.JetService;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+
+import java.util.concurrent.CancellationException;
+
+import static com.hazelcast.jet.pipeline.JournalInitialPosition.START_FROM_OLDEST;
+
+public class JobSuspendResumeListener {
+
+    public static void main(String[] args) throws InterruptedException {
+        // create two instances
+        HazelcastInstance hz1 = Hazelcast.newHazelcastInstance();
+        HazelcastInstance hz2 = Hazelcast.newHazelcastInstance();
+
+        Pipeline p = Pipeline.create();
+        p.readFrom(Sources.<Integer, Integer>mapJournal("source", START_FROM_OLDEST))
+                .withoutTimestamps()
+                .writeTo(Sinks.list("sink"));
+
+        JetService jet1 = hz1.getJet();
+        JobConfig jobConfig = new JobConfig();
+        // job name is optional
+        String jobName = "sampleJob";
+        jobConfig.setName(jobName);
+        Job job = jet1.newJob(p, jobConfig);
+
+        // Register JobStatusListener
+        job.addStatusListener(event ->
+                System.out.printf("Job status changed: %s -> %s. User requested? %b%n",
+                    event.getPreviousStatus(),
+                    event.getNewStatus(),
+                    event.isUserRequested())
+        );
+
+        // printing the job name
+        System.out.println("Job '" + job.getName() + "' is submitted.");
+
+        // we can suspend the job
+        Thread.sleep(1000);
+        System.out.println("Suspending the job...");
+        job.suspend();
```

Comment:
After suspending and resuming, we should wait for `SUSPENDED` and `RUNNING` statuses. Waiting one second may be enough for state transition, but the best practice is polling the status or listening to it.
```java
private static void waitForStatus(Job job, JobStatus status) throws InterruptedException {
    while (job.getStatus() != status) {
        Thread.sleep(100);
    }
}
```
By the way, it is not possible to observe a restart reliably by polling the status, but it can be observed by a listener since it is guaranteed to report `RUNNING` → `NOT_RUNNING` → `STARTING` → `RUNNING` transition. So, you might want to dedicate this sample (or a new one) to that usage, which is I think more interesting than logging the status. You can simply increment a count whenever you receive a `RUNNING` status and wait for a specific run count to infer a number of restarts. However, the unit tests show this particular usage and if they are accessible enough to users, you may skip this. If they aren't, we might consider encouraging users to read them because they usually demonstrate best practices.